odd-collector/odd_collector/adapters/postgresql/repository.py

```diff
@@ -101,11 +109,21 @@ def get_columns(self):
 
     def get_enums(self):
         with self.conn.cursor() as cur:
-            return [EnumTypeLabel(*raw) for raw in self.execute(self.enums_query, cur)]
+            return [
+                EnumTypeLabel(*raw)
+                for raw in self.execute(
+                    self.enums_query(self.get_filtered_schema_names_str), cur
+                )
+            ]
 
     def get_primary_keys(self):
         with self.conn.cursor() as cur:
-            return [PrimaryKey(*raw) for raw in self.execute(self.pks_query, cur)]
+            return [
+                PrimaryKey(*raw)
+                for raw in self.execute(
+                    self.pks_query(self.get_filtered_schema_names_str), cur
+                )
+            ]
```

Comment:
if we cannot use list comprehension inline let's decompose it then for readability
```suggestion
            raw_data = self.execute(
                self.pks_query(self.get_filtered_schema_names_str), cur
            )
            return [PrimaryKey(*raw) for raw in raw_data]
```

Reply:
fixed